typesense-minibar.js

```diff
@@ -1,9 +1,23 @@
 /*! https://github.com/jquery/typesense-minibar 1.0.2 */
 globalThis.tsminibar = function tsminibar (form) {
-  const { origin, key, collection } = form.dataset;
+  const { origin, collection } = form.dataset;
   const group = !!form.dataset.group;
   const cache = new Map();
   const state = { query: '', hits: [], cursor: -1, open: false };
+  const searchParams = new URLSearchParams({
+    per_page: '5',
+    query_by: 'hierarchy.lvl0,hierarchy.lvl1,hierarchy.lvl2,hierarchy.lvl3,hierarchy.lvl4,hierarchy.lvl5,content',
+    include_fields: 'hierarchy.lvl0,hierarchy.lvl1,hierarchy.lvl2,hierarchy.lvl3,hierarchy.lvl4,hierarchy.lvl5,content,url_without_anchor,url,id',
+    highlight_full_fields: 'hierarchy.lvl0,hierarchy.lvl1,hierarchy.lvl2,hierarchy.lvl3,hierarchy.lvl4,hierarchy.lvl5,content',
+    group_by: 'url_without_anchor',
+    group_limit: '1',
+    sort_by: 'item_priority:desc',
+    snippet_threshold: '8',
+    highlight_affix_num_tokens: '12',
+    'x-typesense-api-key': form.dataset.key,
+    ...JSON.parse(form.dataset.searchParams || '{}')
```

Comment:
I'd like to avoid parsing JSON here, as extra overhead. It's tiny, I'm sure, but for this library, given no commercia constraints, I'm trying to demonstrate the best we can do. Even if it's only a small difference!

I think it might also ease adoption/transparancy if the HTML did not have to contain (escaped) JSON, or e.g. having to switch from double quotes to single quotes, which creates extra friction that might make developers like the library less.

In my experience, template libraries and component frameworks, tend to turn embedded JSON into HTML like this:

```html
<div data-params="{&quot;my_option&quot;:&quot;my_value&quot;,&quot;foo&quot;:&quot;bar&quot;}">
```

There's a couple different ways we could format this differently. One option might be to document the `data-search-params` attribute as being a query string. That would take this form:

```html
<div data-params="my_option=my_value&foo=bar">
```

However, that would make it difficult to override existing options (vs enabling other options).

Could you share an example of an option you wanted to add or override?

Reply:
> Could you share an example of an option you wanted to add or override?

Sure, here are my overrides (for now):
```html
data-search-params='{"per_page":50,"sort_by":"_eval(hierarchy.lvl0:=`{{ site.title }}`):desc,item_priority:desc"}'
```
I don't think that these changes are worth including in the defaults or getting a custom setting, generic overriding is probably the right approach here.

> attribute as being a query string

That's a very smart approach! The same overrides encoded as a URL params is:

```html
data-search-params="per_page=50&sort_by=_eval(hierarchy.lvl0:=`Święty na Każdy Dzień`):desc,item_priority:desc"
```

This works great because TypeSense API is designed around URL params anyway, so there's no need for anything as complex as JSON. HTML also seems to handle this form of data well too.

I've pushed a commit switching to URL params.

> avoid parsing JSON here, as extra overhead

Do you think that the overhead will be even measurable? As you said, the amount of data is tiny, and parsing is done only once, when loading a search bar.